### PR TITLE
Adding new composite actions for use in algokit templates

### PR DESCRIPTION
**Pull request by aorumbayev**
# Proposed changes

- Adds `env-to-placeholders` - provides a generic mechanism to create .env file with placeholders (to be replaced later with actual secrets after build is performed by vite or other bundlers)
- Adds `placeholders-to-secrets` - provides a generic mechanism to replace placeholder values with secrets/vars provided to the composite action input

# Further notes

- Same actions can be reused within the repo (can be handled and smoke tested in separate pr)
- These actions have been tested on private repo with instance of react algokit template and they work fine. Example usage below:
```
      - name: Create placeholder .env file
        uses: './.github/env-to-placeholders'
        with:
          env-output-path: './.env'
          env-template-path: './.env.template'
          env-variable-prefix: VITE_

      ... perform build actions

      - name: Replace template vars
        uses: './.github/placeholders-to-secrets'
        with:
          app-artifact-path: './.vercel/output'
          static-site-transforms: |-
            VITE_ALGOD_TOKEN:${{ secrets.VITE_ALGOD_TOKEN }}
            VITE_ALGOD_SERVER:${{ vars.VITE_ALGOD_SERVER }}
            VITE_ALGOD_PORT:${{ vars.VITE_ALGOD_PORT }}
            VITE_ALGOD_NETWORK:${{ vars.VITE_ALGOD_NETWORK }}
            VITE_INDEXER_SERVER:${{ vars.VITE_INDEXER_SERVER }}
            VITE_INDEXER_PORT:${{ vars.VITE_INDEXER_PORT }}
            VITE_INDEXER_TOKEN:${{ secrets.VITE_INDEXER_TOKEN }}
            VITE_ENVIRONMENT:${{ vars.VITE_ENVIRONMENT }}
```

### CODE REVIEW

File: .github/actions/placeholders-to-secrets/action.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Replace placeholders with secrets in build artifacts
+description: Replace placeholder variables in static site build artifacts
+
+inputs:
+  app-artifact-path:
+    description: Path to the app artifact directory
+    required: true
+
+  static-site-transforms:
```
Comment (robdmoore) [inline .github/actions/placeholders-to-secrets/action.yaml:9]: Should this be something more like `placeholder-transforms`?

**Review by robdmoore — APPROVED** [review]

File: .github/actions/placeholders-to-secrets/action.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        VITE_CLIENT_ID:SECRET VALUE
+        VITE_TENANT_ID:SECRET VALUE
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Replace variables in app artifacts
+      run: |
+        ls -a
```
[the remark this answers is not in the document]
Comment (aorumbayev) [reply .github/actions/placeholders-to-secrets/action.yaml:22]: Actually good catch the ls -a was just something i had to test this thing on my private repo, will remove it

File: .github/actions/placeholders-to-secrets/action.yaml (hunk 1 above)
Comment (aorumbayev) [reply .github/actions/placeholders-to-secrets/action.yaml]: @robdmoore done

**Comment by cuzzlor** [comment]
Looks fine to me, I'm interested to see how it's used.

So far with VITE projects I've included the various configs in source, e.g. `.env.dev`, `.env.test` then built for each environment e.g. `vite build --mode dev`.

I'm interested to know when you'd not use that method.

**Pull request merged**